hpss.js

```diff
@@ -201,19 +201,39 @@ var removeid = 0;
 var removed = 0;
 if(config.remove) async.eachSeries(config.remove, function(del, next) {
     if(!del.hpsspath) return next("hpsspath not set for remove request");
-
-    context.rm(del.hpsspath, function(err, out) {
-        var key = ".remove_"+(removeid++);
-        if(err) {
-            console.error(err);
-            progress(key, {status: "failed", msg: "Failed to remove a file:"+del.hpsspath}, function() {
-                next(); //skip this file and continue with other files
+    if (del.directory) {
+	context.rmdir(del.hpsspath, function(err, out) {
+                var key = ".remove_"+(removeid++);
+		var dir_msg = "Directory not empty";
+                if(err && !err.includes(dir_msg)) {
```

Comment:
Can you replace this with 

```
if(err && err.code == 64) {
```

err.code 64 should be raised by hsi if the directory is not empty